src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj

```diff
@@ -5,7 +5,7 @@
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
     <AssemblyName>System.Private.ServiceModel</AssemblyName>
-    <AssemblyVersion>4.0.1.0</AssemblyVersion>
+    <AssemblyVersion>4.1.0.0</AssemblyVersion>
     <RootNamespace>System.ServiceModel</RootNamespace>
```

Comment:
Done - #386. (I've been learning to cherry pick changes from one local branch to another, so that's why this took a little long :smile:) 

This is the all-up issue (PR) that I want the API review to happen on. Once the API review is done for a particular set of changes, I'll split it out and have it in a separate PR. The reason for this is that each individual change doesn't make sense on its own, but makes sense when viewed as a whole.